fideslib/oauth/api/routes/user_endpoints.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from datetime import datetime
+from typing import Optional, Union
+
+from fastapi import APIRouter, Depends, HTTPException, Security
+from fastapi_pagination import Page, Params
+from fastapi_pagination.bases import AbstractPage
+from fastapi_pagination.ext.sqlalchemy import paginate
+from sqlalchemy.orm import Session
+from sqlalchemy_utils import escape_like
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_201_CREATED,
+    HTTP_204_NO_CONTENT,
+    HTTP_400_BAD_REQUEST,
+    HTTP_403_FORBIDDEN,
+    HTTP_404_NOT_FOUND,
+)
+
+from fideslib.core.config import FidesConfig
+from fideslib.models.client import ADMIN_UI_ROOT, ClientDetail
+from fideslib.models.fides_user import FidesUser
+from fideslib.models.fides_user_permissions import FidesUserPermissions
+from fideslib.oauth.api import urn_registry as urls
+from fideslib.oauth.api.deps import get_config, get_db, verify_oauth_client
+from fideslib.oauth.schemas.oauth import AccessToken
+from fideslib.oauth.schemas.user import (
+    UserCreate,
+    UserCreateResponse,
+    UserLogin,
+    UserLoginResponse,
+    UserResponse,
+)
+from fideslib.oauth.scopes import (
+    PRIVACY_REQUEST_READ,
+    USER_CREATE,
+    USER_DELETE,
+    USER_READ,
+)
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@router.post(
+    urls.USERS,
+    dependencies=[Security(verify_oauth_client, scopes=[USER_CREATE])],
+    status_code=HTTP_201_CREATED,
+    response_model=UserCreateResponse,
+)
+def create_user(*, db: Session = Depends(get_db), user_data: UserCreate) -> FidesUser:
+    """Create a user given a username and password."""
+    user_exists = FidesUser.get_by(db, field="username", value=user_data.username)
```

Comment:
I don't really like this name either, but it was the best I could come up with. I'll tell you why I did this and why I used the name to see if you have any better ideas.

`user` gets different types after a `get_by` and `create`. mypy gave an error on the `user = FidesUser.create(db=db, data=user_data.dict())` line since the type was initially set with the `get_by`. Because of this I needed to differentiate the two variable names. I used `user_exists` since I am using it as if it is a boolean.

Reply:
OK, the `TypeVar` also fixed the mypy error here so I was able to change the name back to `user`.